src/App.tsx

```diff
@@ -15,71 +16,104 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string;
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  handleGoodButtonClick = (condition: boolean, nameToSet: string) => {
+    if (condition) {
+      this.setState({
+        selectedGood: '',
+      });
+    } else {
+      this.setState({
+        selectedGood: nameToSet,
+      });
+    }
+  };
```

Comment:
Interesting solution!

Reply:
to make this code clearer, you can pass needed value directly to this function (empty string or good string)
or will be better to create the variable at the start of the function, that will be assigned to the needed value according to some condition, and then call once setState
because for now you repeated some code twice